### ISSUE TITLE
feature request: allow unix socket path to be used instead of TCP port, and allow listening IP addresses to be specified

### ISSUE THREAD

**Issue by toddpfaff** [CARTAvis/carta-node-server#29]
CARTA server defaults to listening on port 8000 on all interfaces.

In many cases this service will be run behind an nginx or apache reverse proxy on the same host.  It is common practice to allow such a service to listen on a unix socket file path instead of a TCP port, and to allow a TCP port listener to be restricted to localhost only.  I'd like to see both of these options available for CARTA.

**Comment by veggiesaurus** [comment]
do you know if this is possible to implement with the NodeJS express server?

**Comment by toddpfaff** [comment]
I don't know about "express" but nodejs [server.listen()](https://nodejs.org/api/net.html#net_server_listen) can use a path or port and other variations.

**Comment by veggiesaurus** [comment]
it looks like express's `listen()` is basically just a wrapper for the nodeJS function. So this should work ok. For the path, we can just expand the schema to allow string values for the `serverPort` field, and we can introduce a new field `listenHost` or `serverHost` for specifying the interface

**Comment by veggiesaurus** [comment]
I've tagged this as Release 2.0, but it can quite easily be added to the 1.4 version, I think. It's a pretty minor adjustment

**Comment by veggiesaurus** [comment]
@toddpfaff I've implemented a fix for this issue in PR #49, and have added a backport to the 1.4 release. If you'd like to test it out, I've attached a package of the PR, produced via `npm pack`. To install it, just unzip it (sorry, github doesn't like tgz files!) and `npm install -g ./carta-controller-1.4.4.tgz`.

The NodeJS `Server` class accepts a `hostname` parameter for restricting the listening interface, but this can only be used in conjunction with a numeric port. If you want to use a unix path, the `serverInterface` field needs to be left blank.

[carta-controller-1.4.4.zip](https://github.com/CARTAvis/carta-controller/files/6107226/carta-controller-1.4.4.zip)
